src/components/map/MapService.js

```diff
@@ -827,6 +828,10 @@ goog.require('ga_urlutils_service');
           }
         };
 
+        this.getLayers = function() {
+          return layersPromise;
+        };
+
```

Comment:
I would not introduce this promise. I would instead create a local promise in backgrounddirective, that will be resolved on the first gaLayersChange event.